lib/puppet/application_support.rb

```diff
@@ -53,6 +53,9 @@ def self.configure_indirector_routes(application_name)
       route_file = Puppet[:route_file]
       if Puppet::FileSystem.exist?(route_file)
         routes = Puppet::Util::Yaml.safe_load_file(route_file, [Symbol])
+        # Reroute master settings to server settings when master is configured in the
+        # route file, and application is not master
+        routes["server"] = routes.delete("master") if routes.key?("master") && application_name != "master"
```

Comment:
I think we also want to handle the case where the routes file contains `server` but the application is [master](https://github.com/puppetlabs/puppetserver/blob/6.x/src/ruby/puppetserver-lib/puppet/server/master.rb). Handling both cases will allow us to switch the puppetlabs-puppetdb module and puppetserver over to using `server` independently. Also maybe just alias the entries in the hash instead of deleting them?

```ruby
if routes["server"] && !routes["master"]
  routes["master"] = routes["server"]
elsif routes["master"] && !routes["server"]
  routes["server"] = routes["master"]
end
```

If the routes file contains both `master` and `server`, then maybe merge them? Or not and warn? Or assume the user knows what they're doing?